test/Microsoft.ML.Benchmarks/Helpers/BaseBenchClass.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using System;
+
+namespace Microsoft.ML.Benchmarks
+{
+    public abstract class BaseBenchClass
+    {
+        public int _passed;
+
+        [IterationCleanup]
+        public void CheckForInnerException()
+        {
+            if (_passed < 0)
+            {
+                throw new Exception($"The benchmark iteration threw an unhandled exception");
```

Comment:
can we get some more info here? any details about the exception? or make Maml just throw instead of returning error codes?

Reply:
@TomFinley is it okay to change it to  maml throwing instead of benchmarks ?

Reply:
It is for various reasons it may not be best to describe here not acceptable for maml itself to throw.